### PR TITLE
chore: refactor AceEditorWrapper to functional component

### PR DESCRIPTION
**Pull request by EugeneTorap**
### SUMMARY

Converted AceEditorWrapper class component into functional component.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by EugeneTorap** [comment]
@jinghua-qa @andrey-zayats Can you test it?

**Comment by andrey-zayats** [comment]
> @jinghua-qa @andrey-zayats Can you test it?
What area can this pr affect? (where should we run regression tests?)

**Comment by EugeneTorap** [comment]
SQL Lab editor

**Comment by andrey-zayats** [comment]
> SQL Lab editor

Okay, I'll take a look

**Comment by EugeneTorap** [comment]
@andrey-zayats If the issue is already in master then we need to create GitHub issue and fix in a separate PR. Right?

**Comment by andrey-zayats** [comment]
> @andrey-zayats If the issue is already in master then we need to create GitHub issue and fix in a separate PR. Right?

Yeap, you're right

**Comment by andrey-zayats** [comment]
PR LGTM, qa verified.

File: superset-frontend/src/SqlLab/components/AceEditorWrapper/AceEditorWrapper.test.tsx [hunk 1]
```diff
@@ -83,6 +73,9 @@ describe('AceEditorWrapper', () => {
   });
 
   it('renders sql from unsaved change', () => {
+    // TODO: need to rewrite the test because we retrieve 'currentQueryEditor' in SqlEditor
```
Comment (michael-s-molina) [inline superset-frontend/src/SqlLab/components/AceEditorWrapper/AceEditorWrapper.test.tsx:76]: @EugeneTorap Can you rewrite the test? It's important to have a test case for this scenario.

Comment (EugeneTorap) [reply superset-frontend/src/SqlLab/components/AceEditorWrapper/AceEditorWrapper.test.tsx:76]: @michael-s-molina Can I do it in the next PR?

**Review by zhaoyongjie — APPROVED** [review]
LGTM! Thanks for your patient and hard work.

**Pull request merged**